### ISSUE TITLE
[🚀 Feature]: Java/CS FirefoxProfile to support addExtensioin temporary

### ISSUE THREAD

**Issue by GregDomjan** [SeleniumHQ/selenium#10084]
### Feature and motivation

Add to `FirefoxProfile::addExtension` option to add extensions as temporary

Other languages (Python, Ruby and Javascript) support providing config of adding extension as temporary 
For some reason Java/C# doesn't provide config, just provides false, or nothing
https://github.com/SeleniumHQ/selenium/blob/trunk/java/src/org/openqa/selenium/firefox/AddHasExtensions.java#L105
https://github.com/SeleniumHQ/selenium/blob/trunk/dotnet/src/webdriver/Firefox/FirefoxDriver.cs#L299

### Usage example

When testing on release (non developer/nightly), an 'unsigned' xpi won't load if it isn't loaded as temporary 
- where localy signed doesn't count as signed.

On windows, when unpacking the xpi to load the folder, can lead to failures because of windows adding content like thumbs.db

```
		FirefoxProfile firefoxProfile = new FirefoxProfile();
		firefoxProfile.addExtension(a, true);
```

**Comment by GregDomjan** [comment]
Python, Ruby and Javascript support, I thought we might try C# as a workaround, also found that it doesn't support temporary option
https://github.com/SeleniumHQ/selenium/blob/trunk/dotnet/src/webdriver/Firefox/FirefoxDriver.cs#L299

**Comment by titusfortner** [comment]
So, this feature won't be part of FirefoxProfile class, but we can add support to the driver's installAddon method.

**Comment by titusfortner** [comment]
Java fixed: 932cbf96052be18e3314c79dc68271446e60c1e3

**Comment by titusfortner** [comment]
Just realized this is a duplicate of #8357. We'll track it there.
